pkg/opensearch/opensearch.go

```diff
@@ -62,7 +63,24 @@ func (ds *OpenSearchDatasource) QueryData(ctx context.Context, req *backend.Quer
 	}
 
 	query := newTimeSeriesQuery(client, req, intervalCalculator)
-	response, err := query.execute()
+	response, err := checkError(query.execute())
+	return response, err
+}
+
+func checkError(response *backend.QueryDataResponse, err error) (*backend.QueryDataResponse, error) {
+	var invalidQueryTypeError invalidQueryTypeError
+	if errors.As(err, &invalidQueryTypeError) {
+		return &backend.QueryDataResponse{
+			Responses: map[string]backend.DataResponse{
+				invalidQueryTypeError.refId: {
+					Error: fmt.Errorf(`invalid queryType, should be Lucene or PPL`),
```

Comment:
I'm not sure its important, but I think we're losing what the queryType actually was here, it looks like the error from `time_series_query.go` returns it as part of the error string